File: scripts/e2e-screenshot.js
#!/usr/bin/env node
// eslint-disable-next-line @typescript-eslint/no-require-imports
const { chromium } = require('playwright');

const BASE = (process.env.PLAYWRIGHT_BASE_URL || 'http://localhost:3000').replace(/\/$/, '');
const OUT = '/tmp';
const DEMO_PASSWORD = process.env.DEMO_PASSWORD?.trim();
if (!DEMO_PASSWORD) {
  throw new Error('DEMO_PASSWORD is required for the authenticated screenshot flow.');
}

async function screenshot(url, file) {
  const browser = await chromium.launch({
    args: ['--no-sandbox', '--disable-gpu', '--disable-dev-shm-usage']
  });
  const page = await browser.newPage();
  try {
    await page.goto(url, { timeout: 15000, waitUntil: 'networkidle' });
    await page.screenshot({ path: file, fullPage: true });
    console.log(`OK: ${url} -> ${file}`);
  } catch (e) {
    console.error(`FAIL: ${url} -> ${e.message}`);
  } finally {
    await browser.close();
  }
}

async function main() {
  // Landing + sign-in
  await screenshot(`${BASE}/ar`, `${OUT}/roastgrid-landing.png`);
  await screenshot(`${BASE}/ar/sign-in`, `${OUT}/roastgrid-signin.png`);

  // Login flow
  const browser = await chromium.launch({
    args: ['--no-sandbox', '--disable-gpu', '--disable-dev-shm-usage']
  });
  const page = await browser.newPage();
  await page.goto(`${BASE}/ar/sign-in`, { timeout: 15000 });
  await page.fill("input[type='email'], input[name='email']", 'admin@roastgrid.app');
  await page.fill("input[type='password']", DEMO_PASSWORD);
  await page.click("button[type='submit']");
  await page.waitForURL('**/dashboard**', { timeout: 10000 });
  await page.screenshot({ path: `${OUT}/roastgrid-dashboard.png`, fullPage: true });
  console.log(`OK: login -> dashboard`);
  await browser.close();

  // Other pages
  await screenshot(`${BASE}/ar/dashboard`, `${OUT}/dashboard.png`);
  await screenshot(`${BASE}/ar/pos`, `${OUT}/pos.png`);
  await screenshot(`${BASE}/ar/inventory`, `${OUT}/inventory.png`);
  await screenshot(`${BASE}/ar/accounting/reports`, `${OUT}/accounting.png`);
}

main().catch(e => { console.error(e); process.exit(1); });

File: scripts/e2e-screenshot.py
#!/usr/bin/env python3
"""E2E screenshot taker for RoastGrid using Playwright + Chromium"""
import asyncio
import os
import sys
from pathlib import Path

async def take_screenshot(url: str, path: str, wait_for: str = None):
    from playwright.async_api import async_playwright
    async with async_playwright() as p:
        browser = await p.chromium.launch(
            args=["--no-sandbox", "--disable-gpu", "--disable-dev-shm-usage"]
        )
        page = await browser.new_page()
        try:
            await page.goto(url, timeout=15000, wait_until="networkidle")
            if wait_for:
                await page.wait_for_selector(wait_for, timeout=5000)
            await page.screenshot(path, full_page=True)
            print(f"OK: {url} -> {path}")
        except Exception as e:
            print(f"FAIL: {url} -> {e}", file=sys.stderr)
        finally:
            await browser.close()

async def main():
    password = os.environ.get("DEMO_PASSWORD", "").strip()
    if not password:
        raise RuntimeError("DEMO_PASSWORD is required for the authenticated screenshot flow.")

    base = os.environ.get("PLAYWRIGHT_BASE_URL", "http://localhost:3000").rstrip("/")
    screens = [
        (f"{base}/ar", "/tmp/roastgrid-landing.png", "[data-testid='sign-in-form'], form, [type='email']"),
        (f"{base}/ar/sign-in", "/tmp/roastgrid-signin.png", "form"),
    ]

    for url, path, _ in screens:
        await take_screenshot(url, path)

    # Login
    from playwright.async_api import async_playwright
    async with async_playwright() as p:
        browser = await p.chromium.launch(
            args=["--no-sandbox", "--disable-gpu", "--disable-dev-shm-usage"]
        )
        page = await browser.new_page()
        await page.goto(f"{base}/ar/sign-in", timeout=15000)
        await page.fill("input[type='email'], input[name='email']", "admin@roastgrid.app")
        await page.fill("input[type='password']", password)
        await page.click("button[type='submit']")
        await page.wait_for_url("**/dashboard**", timeout=10000)
        await page.screenshot(path="/tmp/roastgrid-dashboard.png", full_page=True)
        print("OK: login -> dashboard")
        await browser.close()

    pages = [
        (f"{base}/ar/dashboard", "/tmp/dashboard.png"),
        (f"{base}/ar/pos", "/tmp/pos.png"),
        (f"{base}/ar/inventory", "/tmp/inventory.png"),
        (f"{base}/ar/accounting/reports", "/tmp/accounting.png"),
    ]
    for url, path in pages:
        await take_screenshot(url, path)

asyncio.run(main())
